src/main/java/com/google/devtools/build/lib/metrics/MetricsCollector.java

```diff
@@ -322,17 +353,50 @@ private TimingMetrics finishTimingMetrics() {
     return timingMetrics.build();
   }
 
+  static class DurationAdder {
+
+    final LongAdder duration;
+    final AtomicBoolean valid;
+
+    public DurationAdder() {
+      valid = new AtomicBoolean(true);
+      duration = new LongAdder();
+    }
+
+    public void add(Duration duration) {
+      if (this.valid.get())
```

Comment:
@larsrc-google  Do You think this check makes sense? It might be cheaper to sum all values even if the result is inconsistent than executing his check.

Reply:
It'd waste some memory to not have this check. Summing should definitely have it. I would not micro-optimize without data behind it.

Reply:
@larsrc-google actually, I already pushed a version without that check

Reply:
That's probably slower, now that I think about it. The `.get()` method is probably faster than the `.add()` method. `.add()` needs to do some fancy things to handle multithreading, possibly including acquiring a lock, while `.get()` is non-atomic, which is fine in this case.

Reply:
Since invalidate only ever sets to true, it doesn't need to be compareAndSet. The valid bit is a latch, essentially, but there is no existing data structure for that.

Reply:
Changed it to the simple 'if' check